[bug]: we have duplicate exports (due to the v2 folder)

### What version of strapi-design-system are you using?

all versions

### What's Wrong?

We currently export `Link` from `src/Link` and `src/v2/Link` therefore you can't just do `import { Link } from '@strapi/design-system'`. This isn't just isolated to `Link` but all v2 components.

### To Reproduce

Try `import { Link } from '@strapi/design-system'`.

### Expected Behaviour

To not crash and be able to use the components.

My solution would be to *not* export the `v2` components from `index` and instead _require_ a user to do a direct import of the component. 

I also would advocate we stop doing the `v2` system at all costs.

Just stumbled upon this issue. After update from 1.1.0 to 1.3.1 `import {SideNav} from '@strapi/design-system'` has stopped working. It took some time to figure out what happened. 

I don't see an easy workaround. Some magic in package.json (`exports: {'"./*": "./dist/*}`?) causes direct import to fail:

```javascript
import {  SubNav } from '@strapi/design-system/dist/SubNav'

// webpack error
// node_modules/@strapi/design-system/dist/dist/SubNav doesn't exist
```

also tried this, failed as well (we're using TS)


```javascript
import {  SubNav } from '@strapi/design-system/SubNav'

// TS error
// TS2307: Cannot find module '@strapi/design-system/SubNav' or its corresponding type declarations.
//  12 | import { SubNav } from '@strapi/design-system/SubNav';

```

This worked:

```javascript
const  { SubNav } = require('@strapi/design-system/SubNav')
```

Would be best to maybe ignore v2 in default exports? Currently it looks like this in `@strapi/design-system/dist/index.js`

```javascript

import "./MainNav/MainNav.js";
import "./MainNav/NavBrand.js";
import "./MainNav/NavLink.js";
import "./MainNav/NavSection.js";
import "./MainNav/NavSections.js";
import "./MainNav/NavUser.js";
import "./MainNav/NavCondense.js";
...
import "./v2/SubNav/SubNav.js";
import "./v2/SubNav/SubNavHeader.js";
import "./v2/SubNav/SubNavLink.js";
import "./v2/SubNav/SubNavLinkSection.js";
import "./v2/SubNav/SubNavSection.js";
import "./v2/SubNav/SubNavSections.js";

# no export at all
```

--- a/packages/strapi-design-system/src/index.js
+++ b/packages/strapi-design-system/src/index.js
@@ -1,6 +1,3 @@
-/* eslint-disable import/export */
-// TODO: the import error will be fixed in a different PR
-
 export * from './Accordion';
 export * from './Alert';
 export * from './Avatar';
@@ -8,7 +5,7 @@ export * from './Badge';
 export * from './BaseButton';
 export * from './BaseCheckbox';
 export * from './BaseLink';
-export * from './BaseRadio';
+export { BaseRadio } from './BaseRadio';
 export * from './Box';
 export * from './Breadcrumbs';
 export * from './Button';
@@ -63,5 +60,4 @@ export * from './ToggleInput';
 export * from './Tooltip';
 export * from './Typography';
 export * from './VisuallyHidden';
-export * from './v2';
 export * from './themes';

